test/CellarWithMultiAssetDeposit.t.sol

```diff
@@ -0,0 +1,331 @@
+        priceRouter.addAsset(WBTC, settings, abi.encode(stor), price);
+        price = uint256(mockUsdtUsd.latestAnswer());
+        settings = PriceRouter.AssetSettings(CHAINLINK_DERIVATIVE, address(mockUsdtUsd));
+        priceRouter.addAsset(USDT, settings, abi.encode(stor), price);
+
+        // Add adaptors and ERC20 positions to the registry.
+        registry.trustPosition(usdcPosition, address(erc20Adaptor), abi.encode(USDC));
+        registry.trustPosition(wethPosition, address(erc20Adaptor), abi.encode(WETH));
+        registry.trustPosition(wbtcPosition, address(erc20Adaptor), abi.encode(WBTC));
+        registry.trustPosition(usdtPosition, address(erc20Adaptor), abi.encode(USDT));
+
+        string memory cellarName = "Cellar V0.0";
+        uint256 initialDeposit = 1e6;
+        deal(address(USDC), address(this), initialDeposit);
+        USDC.approve(0xa0Cb889707d426A7A386870A03bc70d1b0697598, initialDeposit);
+        uint64 platformCut = 0.75e18;
+        cellar = new CellarWithMultiAssetDeposit(
+            address(this),
+            registry,
+            USDC,
+            cellarName,
+            "POG",
+            usdcPosition,
+            abi.encode(true),
+            initialDeposit,
+            platformCut,
+            type(uint192).max
+        );
+
+        // Set up remaining cellar positions.
+        cellar.addPositionToCatalogue(wethPosition);
+        cellar.addPosition(0, wethPosition, abi.encode(true), false);
+        cellar.addPositionToCatalogue(wbtcPosition);
+        cellar.addPositionToCatalogue(usdtPosition);
+        cellar.addPosition(0, usdtPosition, abi.encode(true), false);
+
+        cellar.setStrategistPayoutAddress(strategist);
+        vm.label(address(cellar), "cellar");
+        vm.label(strategist, "strategist");
+        // Approve cellar to spend all assets.
+        USDC.approve(address(cellar), type(uint256).max);
+        initialAssets = cellar.totalAssets();
+        initialShares = cellar.totalSupply();
+    }
+
+    // ========================================= HAPPY PATH TEST =========================================
+
+    // Can we accept the Curve LP tokens? I dont see why not.
+
+    function testDeposit(uint256 assets) external {
+        assets = bound(assets, 1e6, 1_000_000_000e6);
+
+        deal(address(USDC), address(this), assets);
+        USDC.safeApprove(address(cellar), assets);
+
+        cellar.deposit(assets, address(this));
+
+        assertEq(
+            cellar.totalAssets(),
+            initialAssets + assets,
+            "Cellar totalAssets should equal initial + new deposit."
+        );
+        assertEq(
+            cellar.totalSupply(),
+            initialAssets + assets,
+            "Cellar totalSupply should equal initial + new deposit."
+        ); // Because share price is 1:1.
+    }
+
+    function testDepositWithAlternativeAsset(uint256 assets) external {
```

Comment:
Do you think it is worth it to test with an asset that is not 1:1? What about native ETH?

Reply:
Actually the 1:1 hard coding was removed, so we are already testing with assets that are not 1:1.

Reply:
We still dont support native ETH. Adding logic to support it, takes up too much contract space.

Reply:
Sg2m